Allow multiple policy areas in policy feedback

ztolley opened the pull request:
Previous the backend API had a field named 'policy_area' when creating a policy feedback. The model has changed to allow multiple areas to be specified, and the field associated with that given a new name.

This change:

[x] Refactors interactions to make it easier to add the change
[x] Uses 'policy_areas' instead of 'policy_area' and makes it support an array of values
[x] Fixes some quirks with form display and error display in a fieldset.

yozzo commented on src/apps/interactions/transformers/interaction-form-body-to-api.js:
```diff
@@ -1,13 +1,18 @@
 
 const { transformDateObjectToDateString } = require('../../transformers')
 
+const castToArrayAndRemoveEmpty = (value) => compact(castArray(value))
```
is there a reason why you're using a function declaration instead of a function expression here?

ztolley replied:
Its just a copy of something used elsewhere and the syntax seemed neat and compact. I may even do away with this and just do it inline

reupen commented:
Something odd happening with the styling?

![image](https://user-images.githubusercontent.com/12693549/41351440-433a9850-6f0e-11e8-84f4-62923369680c.png)

The border is too wide and the font of the error message doesn't look right (although it's squished back down here).

Other add-another fields look fine:

![image](https://user-images.githubusercontent.com/12693549/41351500-75d68fbc-6f0e-11e8-9928-9d2bc084c016.png)

reupen commented on src/apps/interactions/transformers/interaction-form-body-to-api.js:
```diff
@@ -0,0 +1,19 @@
+
+const { transformDateObjectToDateString } = require('../../transformers')
+
+const castToArrayAndRemoveEmpty = (value) => compact(castArray(value))
```
This is also defined in `src/apps/events/transformers.js` and `src/apps/investment-projects/middleware/forms/requirements.js` also does `compact(castArray(...))`. Can the function be defined somewhere common and then used consistently?

ztolley replied:
I only didn't do that because it's a single line and felt like overkill, but can do

ztolley replied:
I may just put this inline tbh

ztolley commented:
That's strange because they all use the same components, no styling was done at all for this. I'll take a look in case there is something I'm missing but not sure what.

reupen commented:
It seems like the wrong border size might be because there's no left nav on the page.

As for the error message font and placement, the client contacts field mentioned doesn't seem to be using the same macro.

yeahfro commented on test/acceptance/pages/interactions/interaction.js:
```diff
@@ -44,7 +44,7 @@ module.exports = {
     event: '#field-event',
     policyIssueType: '#field-policy_issue_type',
-    policyArea: '#field-policy_area',
+    policyArea: '#field-policy_areas',
```
Hello @ztolley 😃 

Strictly speaking this should belong to the commit before because the build is broken until this commit is applied.

ztolley replied:
Github has done its usual thing of getting the commit order wrong after a rebase, but you are correct. I'll merge those two. Welcome back, we missed you.

ztolley replied:
But, generally, I'm of the mind that it's ok to have a commit that fails tests, the idea is you have a branch that must pass them. In some TDD shops the first thing they do is create and commit tests that will fail and then say the feature it done when all the tests pass.

ztolley commented:
![capture](https://user-images.githubusercontent.com/56056/41414378-b585a0a6-6fdd-11e8-8f0a-2fe9bccbf724.PNG)

[x] Fixed the style issues in the form
[x] Merged the commit with the fixes for acceptance tests
[x] Refactored casting to compact arrays to a shared function

reupen commented:
Looks better, except that one of the acceptance tests failed (around contact error messages).

ztolley commented:
Yup looking at that now.
